charts/dvpe-deployment-gloo/templates/secret.yaml

```diff
@@ -33,6 +33,26 @@ spec:
 
 {{- end }}
 
+{{- if and .Values.externalSecrets.oidc.key .Values.gloo.authConfig.spec.configs.clientCredentialsPlugin.config.clientId }}
+---
+apiVersion: 'kubernetes-client.io/v1'
+kind: ExternalSecret
+metadata:
+  name: {{ $serviceName }}-oidc-m2m-secrets
```

Comment:
We should use the name `clientCredentials` flow instead of `m2m` to be consistent with the other namings.
I would also replace `oidc` with `oauth2` because the client credentials flow is not part of OIDC. It is a basic OAuth2 flow

Reply:
done

Reply:
`clientCredentials` does not work - only lower case letters are allowed -> `client-credentials`